src/solvers/dg_structured/dg_2d.jl

```diff
@@ -0,0 +1,137 @@
+function rhs!(du::AbstractArray{<:Any,4}, u, t,
+    mesh::StructuredMesh, equations,
+    initial_condition, boundary_conditions, source_terms,
+    dg::DG, cache)
+  # Reset du
+  @timeit_debug timer() "reset ∂u/∂t" du .= zero(eltype(du))
+
+  # Calculate volume integral
+  @timeit_debug timer() "volume integral" calc_volume_integral!(du, u, mesh, equations,
+                                                                dg.volume_integral, dg, cache)
+
+  # Calculate interface fluxes
+  @timeit_debug timer() "interface flux" calc_interface_flux!(u, mesh, equations, dg, cache)
+
+  # Calculate surface integrals
+  @timeit_debug timer() "surface integral" calc_surface_integral!(du, equations, dg, cache)
+
+  # Apply Jacobian from mapping to reference element
+  @timeit_debug timer() "Jacobian" apply_jacobian!(du, mesh, equations, dg, cache)
+
+  # Calculate source terms
+  @timeit_debug timer() "source terms" calc_sources!(du, u, t, source_terms, equations, dg, cache)
+
+  return nothing
+end
+
+
+function calc_volume_integral!(du::AbstractArray{<:Any,4}, u, mesh::StructuredMesh, equations,
+                               volume_integral::VolumeIntegralWeakForm, dg::DGSEM, cache)
+  @unpack derivative_dhat = dg.basis
+
+  @threaded for element in eachelement(dg, cache)
+    for j in eachnode(dg), i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, j, element)
+
+      flux1 = transformed_flux(u_node, 1, mesh, equations)
+      for ii in eachnode(dg)
+        integral_contribution = derivative_dhat[ii, i] * flux1
+        add_to_node_vars!(du, integral_contribution, equations, dg, ii, j, element)
+      end
+
+      flux2 = transformed_flux(u_node, 2, mesh, equations)
+      for jj in eachnode(dg)
+        integral_contribution = derivative_dhat[jj, j] * flux2
+        add_to_node_vars!(du, integral_contribution, equations, dg, i, jj, element)
+      end
+    end
+  end
+
+  return nothing
+end
+
+
+function calc_interface_flux!(u::AbstractArray{<:Any,4}, mesh::StructuredMesh{2},
+                              equations, dg::DG, cache)
+  @unpack elements = cache
+
+  @threaded for element in eachelement(dg, cache)
+    # Interfaces in negative directions
+    # Faster version of "for orientation in (1, 2)"
+    Base.Cartesian.@nexprs 2 orientation->calc_interface_flux!(elements.surface_flux_values, 
+                                                               elements.left_neighbors[orientation, element], 
+                                                               element, orientation, u, mesh, equations, dg)
```

Comment:
```suggestion
    # Interfaces in x-direction (`orientation` = 1)
    calc_interface_flux!(elements.surface_flux_values, elements.left_neighbors[1, element], element, 1, u, mesh, equations, dg)
    
    # Interfaces in x-direction (`orientation` = 2)
    calc_interface_flux!(elements.surface_flux_values, elements.left_neighbors[2, element], element, 2, u, mesh, equations, dg)
```
I know the temptation to use something like `@nexpr`, but it makes the code harder to read, especially for new users. Therefore, I strongly prefer to manually unroll the loop here. The comment `# Faster version of...` should definitely remain, since it explains why you did not use a loop in the first place.

Reply:
The same applies to 3D of course...